### ISSUE TITLE
onEntry firing on nested transition in hierarchical setting

### ISSUE THREAD

**Issue by nenti** [davidkpiano/xstate#49]
## Bug
In a hierarchical state graph onEntry action is fired on a solely nested transition although it is not defined in the nested graph.

## Expected result:
onEntry only fires on entry to a state or on transition to itself but not on a transition of a nested sub graph. 

## Actual result:
onEntry action fires on every transition of nested state graphs. 

## Example:
This [example](https://codepen.io/nenti/pen/WdBpRd) generates a nested `fooBar` machine inside the `ping` state of a `pingPong` state machine.

``` js
const pingPong = Machine({
  initial: 'ping',
  states: {
    ping: {
      onEntry: ['entryEvent'], 
      on: {
        TICK: 'pong'
      },
      initial: 'foo',
      states: {
        foo: {
          on: {
            TACK: 'bar'
          }
        },
        bar: {
          on: {
            TACK: 'foo'
          }
        },
      },
    },
    pong: {
      on: {
        TICK: 'ping'
      }
    },
  }
});
```

Transition from nested state `ping.foo` to `ping.bar` on `TACK` fires `entryEvent` of the `pingPong` machine. Although we stay inside of the state and thus should not have the `entryEvent` fired. 

``` js
pingPong.transition('ping.foo', 'TACK')
// => State {
//   value: { ping: 'bar' },
//   actions: [
//     'entryEvent' // WHY?! 
//   ]
// }
```

**Comment by lmatteis** [comment]
Can you provide an example of the state and the transition output?

**Comment by nenti** [comment]
I don't habe my development environment available so typed this example untested. 

```
const fooBar = {
  initial: 'foo',
  states: {
    foo: {
      on: {
        TACK: 'bar'
      }
    },
    bar: {
      on: {
        TACK: 'foo'
      }
    },
  }
};

const pingPong = Machine({
  initial: 'ping',
  states: {
    ping: {
      onEntry: ['entryEvent'], 
      on: {
        TICK: 'pong'
      }
     ... fooBar,
    },
    pong: {
      on: {
        TICK: 'ping'
      }
    },
  }
});
```

```
transition('ping.foo', 'TACK').value
// => State {
//   value: 'ping.bar',
//   actions: [
//     'entryEvent' // WHY?! 
//   ]
// }
```

Do you get the issue?

**Comment by davidkpiano** [comment]
Published in v3.0.3

**Comment by nenti** [comment]
Updated the version of the codepen and it looks great. Thank you for quick fix!
